workload/terraform/modules/mgmtvm/main.tf

```diff
@@ -0,0 +1,144 @@
+  provision_vm_agent         = true
+  admin_username             = var.local_admin_username
+  admin_password             = var.localpassword
+  encryption_at_host_enabled = true //'Microsoft.Compute/EncryptionAtHost' feature is must be enabled in the subscription for this setting to work https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-host-based-encryption-portal?tabs=azure-powershell
+  secure_boot_enabled        = true
+  vtpm_enabled               = true
+  os_disk {
+    name                 = lower(var.prefix)
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+  # To use marketplace image, uncomment the following lines and comment the source_image_id line
+  source_image_reference {
+    publisher = var.publisher
+    offer     = var.offer
+    sku       = var.sku
+    version   = "latest"
+  }
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.stguai.id
+    ]
+  }
+}
+
+# Resource block to join the MGMT VM to a domain
+resource "azurerm_virtual_machine_extension" "domain_join" {
+  name                       = "${var.prefix}-domainJoin"
+  virtual_machine_id         = azurerm_windows_virtual_machine.mgmt_vm.id
+  publisher                  = "Microsoft.Compute"
+  type                       = "JsonADDomainExtension"
+  type_handler_version       = "1.3"
+  auto_upgrade_minor_version = true
+
+  settings = <<SETTINGS
+    {
+      "Name": "${var.domain_name}",
+      "OUPath": "${var.ou_path}",
+      "User": "${var.domain_user}@${var.domain_name}",
+      "Restart": "true",
+      "Options": "3"
+    }
+SETTINGS
+
+  protected_settings = <<PROTECTED_SETTINGS
+    {
+      "Password": "${var.domain_password}"
+    }
+PROTECTED_SETTINGS
+
+  lifecycle {
+    ignore_changes = [settings, protected_settings]
+  }
+}
+
+resource "azurerm_virtual_machine_extension" "dscStorageScript" {
+  name                 = "AzureFilesDomainJoin"
+  virtual_machine_id   = azurerm_windows_virtual_machine.mgmt_vm.id
+  publisher            = "Microsoft.Compute"
+  type                 = "CustomScriptExtension"
+  type_handler_version = "1.10"
+
+  settings = jsonencode({
+    fileUris         = [var.baseScriptUri]
+    commandToExecute = "powershell.exe -ExecutionPolicy Unrestricted -File ${var.vfile} -DscPath ${var.dsc_storage_path} -StorageAccountName ${var.storage_account_name} -StorageAccountRG ${var.storage_account_rg} -StoragePurpose fslogix -DomainName ${var.domain_name} -IdentityServiceProvider ${var.IdentityServiceProvider} -AzureCloudEnvironment ${var.azure_cloud_environment} -SubscriptionId ${var.workloadSubsId} -DomainAdminUserName ${var.domain_user}@${var.domain_name} -CustomOuPath ${var.storageCustomOuPath} -OUName ${var.ouStgPath} -CreateNewOU ${var.createOuForStorageString} -ShareName ${var.file_share_name} -ClientId ${var.clientid} -DomainAdminUserPassword ${var.domain_password} -verbose"
```

Comment:
variable with the name "storage_account_rg" has not been declare
variable with the name "workloadSubsId" has not been declared
variable with the name "storageCustomOuPath" has not been declared
variable with the name "ouStgPath" has not been declared
variable with the name "createOuForStorageString" has not been declared
variable with the name "file_share_name" has not been declared

Reply:
addressed